sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -543,7 +543,7 @@ public void Dispose()
                     endpoint.AbsoluteUri,
```

Comment:
I was unsuccessful in a few attempts to devise tests that could deterministically reproduce the race condition and validate, without artificially injecting the `ObjectDisposedException`, which didn't offer any real value.

If you've got ideas on an approach, I'm all ears.

Reply:
I don't have a detailed understanding of how the `AmqpConnectionScope` type operates, so it may not make sense, but couldn't we override the `CreateAuthorizationRefreshHandler` with something like:
```csharp
protected override TimerCallback CreateAuthorizationRefreshHandler(...)
{
    // Call Dispose here.
    return base.CreateAuthorizationRefreshHandler(...);
}
```

Reply:
Okay. Just thought about it and it may not work. What about returning an anonymous function that calls dispose and, after that, calls the base implementation?

Reply:
Clever approach; that did the trick.   Thanks!